app/response_svc.py

```diff
@@ -148,3 +153,30 @@ async def _apply_adversary_config(self):
     async def _save_configurations(self):
         with open('plugins/response/conf/response.yml', 'w') as config:
             config.write(yaml.dump(self.get_config(name='response')))
+
+    def _filter_ability_facts(self, unique_facts, relationships, red_pid, original_pid):
+        ability_facts = []
+        for fact in unique_facts:
+            if fact.trait == 'host.process.guid':
+                if self._is_child_guid(relationships, red_pid, original_pid, fact):
+                    ability_facts.append(fact)
+            elif fact.trait == 'host.process.parentguid':
+                if self._is_red_agent_guid(relationships, red_pid, fact):
+                    ability_facts.append(fact)
+            else:
+                ability_facts.append(fact)
+        return ability_facts
+
+    @staticmethod
+    def _is_child_guid(relationships, red_pid, original_pid, fact):
+        for r in relationships:
+            if r.edge == 'has_parentid' and \
+                    (r.target.value.strip() == red_pid or r.target.value.strip() == original_pid) \
+                    and r.source.value == fact.value:
+                return True
+        return False
+
+    @staticmethod
+    def _is_red_agent_guid(relationships, red_pid, fact):
+        red_guid = [r.target.value for r in relationships if r.source.value.strip() == red_pid].pop()
+        return fact.value == red_guid
```

Comment:
Not sure if it would make more sense to create something like a new planner to put this filtering in instead of having it in the response_svc.